iib/workers/tasks/build.py

```diff
@@ -158,34 +158,41 @@ def _finish_request_post_build(
         index image.
     :param str overwrite_from_index_token: the token used for overwriting the input
         ``from_index`` image.
+    :return: pull spec of the index_image
+    :rtype: str
     :raises IIBError: if the manifest list couldn't be created and pushed
     """
     conf = get_worker_config()
+    state_reason = ''
     if from_index and overwrite_from_index:
         log.info(f'Ovewriting the index image {from_index} with {output_pull_spec}')
         index_image = from_index
         exc_msg = f'Failed to overwrite the input from_index container image of {index_image}'
         args = [f'docker://{output_pull_spec}', f'docker://{index_image}']
         _skopeo_copy(*args, copy_all=True, dest_token=overwrite_from_index_token, exc_msg=exc_msg)
+        state_reason = f'Overwrote the index image {from_index} with {output_pull_spec}'
     elif conf['iib_index_image_output_registry']:
         index_image = output_pull_spec.replace(
             conf['iib_registry'], conf['iib_index_image_output_registry'], 1
         )
-        log.info(
-            'Changed the index_image pull specification from %s to %s',
-            output_pull_spec,
-            index_image,
+        output_message = (
+            f'Changed the index_image pull specification from {output_pull_spec} to {index_image}'
         )
+        log.info(output_message)
+        state_reason = output_message
     else:
         index_image = output_pull_spec
+        state_reason = f'Updated the index image with {output_pull_spec}'
 
     payload = {
         'arches': list(arches),
         'index_image': index_image,
-        'state': 'complete',
-        'state_reason': 'The request completed successfully',
+        'state': 'in_progress',
+        'state_reason': state_reason,
     }
-    update_request(request_id, payload, exc_msg='Failed setting the index image on the request')
+    update_request(request_id, payload, exc_msg='Failed updating the index image on the request')
```

Comment:
Why was the exception message changed?

Reply:
Aah, this was from my initial commit, changing it back